sled-agent/src/services.rs

```diff
@@ -1194,6 +1197,155 @@ impl ServiceManager {
                 let running_zone = RunningZone::boot(installed_zone).await?;
                 return Ok(running_zone);
             }
+            ZoneType::Nexus => {
+                let (internal_ip, external_tls) = request
+                    .zone
+                    .services
+                    .iter()
+                    .find_map(|s| {
+                        if let ServiceType::Nexus {
+                            internal_ip,
+                            external_tls,
+                            ..
+                        } = s.details
+                        {
+                            Some((internal_ip, external_tls))
+                        } else {
+                            None
+                        }
+                    })
+                    .ok_or_else(|| {
+                        Error::BadRequest(
+                            "Nexus zone missing Nexus service".to_string(),
+                        )
+                    })?;
+
+                let sled_info = self
+                    .inner
+                    .sled_info
+                    .get()
+                    .ok_or(Error::SledAgentNotReady)?;
+
+                // While Nexus will be reachable via `external_ip`, it communicates
+                // atop an OPTE port which operates on a VPC private IP. OPTE will
+                // map the private IP to the external IP automatically.
+                let port =
+                    installed_zone.opte_ports().nth(0).ok_or_else(|| {
+                        Error::BadRequest(
+                            "Nexus zone missing OPTE port".to_string(),
+                        )
+                    })?;
+                let opte_interface = port.vnic_name();
+                let opte_gateway = &port.gateway().ip().to_string();
```

Comment:
This bit is new, we're pulling out these values instead of calling `ensure_address_for_port` (which was previously performed with a bunch of `zlogin` operations). Instead, these addresses should be set up in the method scripts.